spyder/app/mainwindow.py

```diff
@@ -223,6 +223,22 @@ def get_focus_python_shell():
         return widget.shell
 
 
+def get_environment_infostr():
+    """Find if running inside a conda and/or venv
+    Return None if not the case or anything fails"""
+    debug_print("Collecting environment information")
+    try:
+        try:
+            return "env: {}".format(os.path.basename(sys.prefix))
+        except AttributeError:
+            return ""
+    except Exception as e:
+        debug_print(
+        "Exception {} while trying to obtain environment information".format(e)
+        )
+
+    return ""

```

Comment:
You say "Return None if not the case or anything fails" but it actually returns an empty string.